src/services/refactors/extractType.ts

```diff
@@ -130,17 +130,17 @@ namespace ts.refactor {
     }
 
     function collectTypeParameters(checker: TypeChecker, selection: TypeNode, statement: Statement, file: SourceFile): TypeParameterDeclaration[] | undefined {
-        const result: TypeParameterDeclaration[] = [];
-        return visitor(selection) ? undefined : result;
+        const result = new Set<TypeParameterDeclaration>();
+        return visitor(selection) ? undefined : arrayFrom(result.values());
 
         function visitor(node: Node): true | undefined {
             if (isTypeReferenceNode(node)) {
                 if (isIdentifier(node.typeName)) {
                     const symbol = checker.resolveName(node.typeName.text, node.typeName, SymbolFlags.TypeParameter, /* excludeGlobals */ true);
                     if (symbol) {
                         const declaration = cast(first(symbol.declarations), isTypeParameterDeclaration);
-                        if (rangeContainsSkipTrivia(statement, declaration, file) && !rangeContainsSkipTrivia(selection, declaration, file)) {
-                            result.push(declaration);
+                        if (rangeContainsSkipTrivia(statement, declaration, file) && !rangeContainsSkipTrivia(selection, declaration, file) && !result.has(declaration)) {

```

Comment:
If you're just de-duping declarations you could use `result.indexOf(declaration) > 0` if you want to avoid declaring a Set which will ultimately be converted back to an array anyway.